nima_core/dream_consolidation.py

```
@@ -0,0 +1,542 @@
+    for p in patterns:
+        if p["type"] == "domain_cluster" and p["count"] >= PATTERN_THRESHOLD:
+            keywords_str = ", ".join(p["keywords"][:4])
+            text = (
+                f"Pattern: {p['count']} memories clustered in "
+                f"'{p['domain']}' domain. "
+                f"Core concepts: {keywords_str}. "
+                f"This domain is receiving consistent attention — likely important."
+            )
+            score = min(0.95, 0.5 + p["count"] * 0.06)
+            insights.append({
+                "id": hashlib.md5(text.encode()).hexdigest()[:12],
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)